nodebuilder/node/node.go

```diff
@@ -0,0 +1,110 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/filecoin-project/go-jsonrpc/auth"
+	logging "github.com/ipfs/go-log/v2"
+)
+
+var (
+	buildTime       string
+	lastCommit      string
+	semanticVersion string
+)
+
+// Module defines the API related to interacting with the "administrative"
+// node.
+//
+//go:generate mockgen -destination=mocks/api.go -package=mocks . Module
+type Module interface {
+	// Type returns the node type.
+	Type() Type
+	// Version returns information about the current binary build.
+	Version() Version
+
+	// LogLevelSet sets the given component log level to the given level.
+	LogLevelSet(name, level string) error
+
+	// AuthVerify returns the permissions assigned to the given token.
+	AuthVerify(ctx context.Context, token string) ([]auth.Permission, error)
+	// AuthNew signs and returns a new token with the given permissions.
+	AuthNew(ctx context.Context, perms []auth.Permission) ([]byte, error)
+}
+
+type API struct {
+	Internal struct {
+		Type        func() Type                                                        `perm:"admin"`
+		Version     func() Version                                                     `perm:"admin"`
+		LogLevelSet func(name, level string) error                                     `perm:"admin"`
+		AuthVerify  func(ctx context.Context, token string) ([]auth.Permission, error) `perm:"admin"`
+		AuthNew     func(ctx context.Context, perms []auth.Permission) ([]byte, error) `perm:"admin"`
+	}
+}
+
+func (api *API) Type() Type {
+	return api.Internal.Type()
+}
+
+func (api *API) Version() Version {
+	return api.Internal.Version()
+}
+
+func (api *API) LogLevelSet(name, level string) error {
+	return api.Internal.LogLevelSet(name, level)
+}
+
+func (api *API) AuthVerify(ctx context.Context, token string) ([]auth.Permission, error) {
+	return api.Internal.AuthVerify(ctx, token)
+}
+
+func (api *API) AuthNew(ctx context.Context, perms []auth.Permission) ([]byte, error) {
+	return api.Internal.AuthNew(ctx, perms)
+}
+
+type admin struct {
```

Comment:
can you move this struct and its methods to a new file

Reply:
done